### ISSUE TITLE
Tiller service accounts are not created as expected

### ISSUE THREAD

**Issue by ironhalo** [Praqma/helmsman#125]
Helmsman doesn't create tiller service accounts:
```yaml
---
metadata:
  mcs: "REDACTED"
  maintainer: "REDACTED"

settings:
  storageBackend: "secret"
  kubeContext: "REDACTED"

namespaces:
  kube-system:
    installTiller: true
  dev:
    installTiller: true

helmRepos:
  stable: "https://kubernetes-charts.storage.googleapis.com"

apps:
  metrics-server:
    name: "metrics-server"
    description: "metrics-server"
    namespace: "kube-system"
    enabled: true
    chart: "stable/metrics-server"
    version: "2.0.3"
    valuesFile: ".values.yaml"
    purge: false
    test: false
    protected: false
    wait: true
```

```
helmsman --apply --debug --verbose -f dsf.yaml
 _          _
| |        | |
| |__   ___| |_ __ ___  ___ _ __ ___   __ _ _ __
| '_ \ / _ \ | '_ ` _ \/ __| '_ ` _ \ / _` | '_ \
| | | |  __/ | | | | | \__ \ | | | | | (_| | | | |
|_| |_|\___|_|_| |_| |_|___/_| |_| |_|\__,_|_| |_| version: v1.7.0
A Helm-Charts-as-Code tool.


2018/11/07 11:45:46 INFO: checking Helm version
2018/11/07 11:45:47 INFO: checking kubectl version
2018/11/07 11:45:47 VERBOSE: kubectl client version: v1.12.2
2018/11/07 11:45:47 VERBOSE: Helm client version: v2.11.0+g2e55dbe
2018/11/07 11:45:47 INFO: validating that kubectl is installed.
2018/11/07 11:45:47 INFO: validating that helm is installed.
2018/11/07 11:45:47 INFO: validating that diff is installed.
2018/11/07 11:45:47 INFO: Parsed YAML [[ dsf.yaml ]] successfully and found [ 1 ] apps.
2018/11/07 11:45:47 INFO: namespace validation -- Tiller is desired to be deployed WITHOUT TLS in namespace [ kube-system ].
2018/11/07 11:45:47 INFO: namespace validation -- Tiller is desired to be deployed WITHOUT TLS in namespace [ dev ].
2018/11/07 11:45:47 INFO: setting kubectl context to [ REDACTED ]
2018/11/07 11:45:47 VERBOSE: kubectl config use-context REDACTED
2018/11/07 11:45:47 INFO: creating namespace  kube-system
2018/11/07 11:45:47 VERBOSE: kubectl create namespace kube-system
2018/11/07 11:45:47 WARN: I could not create namespace [ kube-system ]. It already exists. I am skipping this.
2018/11/07 11:45:47 INFO: creating namespace  dev
2018/11/07 11:45:47 VERBOSE: kubectl create namespace dev
2018/11/07 11:45:48 INFO: deploying Tiller in namespace [ kube-system ].
2018/11/07 11:45:48 INFO: initializing helm on the current context and upgrading Tiller on namespace [ kube-system ].
2018/11/07 11:45:48 VERBOSE: helm init --upgrade  --tiller-namespace kube-system --override 'spec.template.spec.containers[0].command'='{/tiller,--storage=secret}'
2018/11/07 11:45:48 INFO: deploying Tiller in namespace [ dev ].
2018/11/07 11:45:48 INFO: initializing helm on the current context and upgrading Tiller on namespace [ dev ].
2018/11/07 11:45:48 VERBOSE: helm init --upgrade  --tiller-namespace dev --override 'spec.template.spec.containers[0].command'='{/tiller,--storage=secret}'
2018/11/07 11:45:49 INFO: checking if helm Tiller is ready in namespace [ kube-system ].
2018/11/07 11:45:49 VERBOSE: helm list --tiller-namespace kube-system
2018/11/07 11:45:49 INFO: waiting for helm Tiller to be ready in namespace [kube-system] ...
2018/11/07 11:45:54 INFO: checking if helm Tiller is ready in namespace [ kube-system ].
2018/11/07 11:45:54 VERBOSE: helm list --tiller-namespace kube-system
2018/11/07 11:45:55 INFO: waiting for helm Tiller to be ready in namespace [kube-system] ...
2018/11/07 11:46:00 INFO: checking if helm Tiller is ready in namespace [ kube-system ].
2018/11/07 11:46:00 VERBOSE: helm list --tiller-namespace kube-system
2018/11/07 11:46:00 ERROR: while waiting for helm Tiller to be ready in namespace [ kube-system ] : Error: secrets is forbidden: User "system:serviceaccount:kube-system:default" cannot list secrets in the namespace "kube-system"
```

**Comment by sami-alajrami** [comment]
@ironhalo you need to add `tillerServiceAccount: "serviceaccountname"` into your namespaces definitions. Helmsman will use that service account if it exists, otherwise, it will create it for you.
More details: https://github.com/Praqma/helmsman/blob/master/docs/how_to/define_namespaces.md#deploying-tiller-into-namespaces
https://github.com/Praqma/helmsman/blob/master/docs/desired_state_specification.md#namespaces

**Comment by ironhalo** [comment]
@sami-alajrami thanks, I'll take a look!

**Comment by ironhalo** [comment]
Next question, it looks like a service account in `kube-system` isn't being bound to the `cluster-admin` clusterrole.  Is there another flag that should be set?

```yaml
 ---
 metadata:
   mcs: ""
   maintainer: ""

 settings:
   storageBackend: "secret"
   kubeContext: ""

 namespaces:
   kube-system:
     installTiller: true
     tillerServiceAccount: "tiller-kube-system"

 helmRepos:
   stable: "https://kubernetes-charts.storage.googleapis.com"

 apps:
   metrics-server:
     name: "metrics-server"
     description: "metrics-server"
     namespace: "kube-system"
     enabled: true
     chart: "stable/metrics-server"
     version: "2.0.3"
     valuesFile: "./metrics-server.yaml"
     purge: false
     test: false
     protected: false
     wait: true
```

```
helmsman --verbose --debug --apply -f helmsman/dsf.yaml
 _          _
| |        | |
| |__   ___| |_ __ ___  ___ _ __ ___   __ _ _ __
| '_ \ / _ \ | '_ ` _ \/ __| '_ ` _ \ / _` | '_ \
| | | |  __/ | | | | | \__ \ | | | | | (_| | | | |
|_| |_|\___|_|_| |_| |_|___/_| |_| |_|\__,_|_| |_| version: v1.7.0
A Helm-Charts-as-Code tool.


2018/11/13 13:32:47 INFO: checking Helm version
2018/11/13 13:32:47 INFO: checking kubectl version
2018/11/13 13:32:47 VERBOSE: kubectl client version: v1.12.2
2018/11/13 13:32:47 VERBOSE: Helm client version: v2.11.0+g2e55dbe
2018/11/13 13:32:47 INFO: validating that kubectl is installed.
2018/11/13 13:32:47 INFO: validating that helm is installed.
2018/11/13 13:32:47 INFO: validating that diff is installed.
2018/11/13 13:32:48 INFO: Parsed YAML [[ helmsman/dsf.yaml ]] successfully and found [ 1 ] apps.
2018/11/13 13:32:48 INFO: namespace validation -- Tiller is desired to be deployed WITHOUT TLS in namespace [ kube-system ].
2018/11/13 13:32:48 INFO: setting kubectl context to [ CONTEXT ]
2018/11/13 13:32:48 VERBOSE: kubectl config use-context CONTEXT
2018/11/13 13:32:48 INFO: creating namespace  kube-system
2018/11/13 13:32:48 VERBOSE: kubectl create namespace kube-system
2018/11/13 13:32:48 WARN: I could not create namespace [ kube-system ]. It already exists. I am skipping this.
2018/11/13 13:32:48 INFO: deploying Tiller in namespace [ kube-system ].
2018/11/13 13:32:48 INFO: validating if serviceaccount [ tiller-kube-system ] exists in namespace [ kube-system ].
2018/11/13 13:32:48 VERBOSE: kubectl get serviceaccount tiller-kube-system -n kube-system
2018/11/13 13:32:48 INFO: service account [ tiller-kube-system ] does not exist in namespace [ kube-system ] .. attempting to create it ...
2018/11/13 13:32:48 INFO: creating service account [ tiller-kube-system ] in namespace [ kube-system ]
2018/11/13 13:32:48 VERBOSE: kubectl create serviceaccount -n kube-system tiller-kube-system
2018/11/13 13:32:49 INFO: creating role [helmsman-tiller] in namespace [ kube-system ]
2018/11/13 13:32:49 VERBOSE: kubectl apply -f temp-modified-role.yaml
2018/11/13 13:32:49 INFO: cleaning up ... deleting temp-modified-role.yaml
2018/11/13 13:32:49 INFO: creating rolebinding for service account [ tiller-kube-system ] in namespace [ kube-system ] with role: helmsman-tiller
2018/11/13 13:32:49 VERBOSE: kubectl create rolebinding tiller-kube-system-binding --role=helmsman-tiller --serviceaccount kube-system:tiller-kube-system -n kube-system
2018/11/13 13:32:50 INFO: initializing helm on the current context and upgrading Tiller on namespace [ kube-system ].
2018/11/13 13:32:50 VERBOSE: helm init --upgrade --service-account tiller-kube-system --tiller-namespace kube-system --override 'spec.template.spec.containers[0].command'='{/tiller,--storage=secret}'
2018/11/13 13:32:50 INFO: checking if helm Tiller is ready in namespace [ kube-system ].
2018/11/13 13:32:50 VERBOSE: helm list --tiller-namespace kube-system
2018/11/13 13:32:50 INFO: waiting for helm Tiller to be ready in namespace [kube-system] ...
2018/11/13 13:32:55 INFO: checking if helm Tiller is ready in namespace [ kube-system ].
2018/11/13 13:32:55 VERBOSE: helm list --tiller-namespace kube-system
2018/11/13 13:32:56 INFO: waiting for helm Tiller to be ready in namespace [kube-system] ...
2018/11/13 13:33:01 INFO: checking if helm Tiller is ready in namespace [ kube-system ].
2018/11/13 13:33:01 VERBOSE: helm list --tiller-namespace kube-system
2018/11/13 13:33:01 INFO: adding repo stable
2018/11/13 13:33:01 VERBOSE: helm repo add stable "https://kubernetes-charts.storage.googleapis.com"
2018/11/13 13:33:02 INFO: updating helm repos
2018/11/13 13:33:02 VERBOSE: helm repo update
2018/11/13 13:33:12 INFO: validating if chart stable/metrics-server-2.0.3 is available in the defined repos.
2018/11/13 13:33:12 VERBOSE: helm search stable/metrics-server --version "2.0.3" -l
2018/11/13 13:33:13 INFO: checking what I need to do for your charts ...
2018/11/13 13:33:13 INFO: mapping the current helm state ...
2018/11/13 13:33:13 INFO: listing all existing releases in namespace [ kube-system ]...
2018/11/13 13:33:13 VERBOSE: helm list --all --max 0 --output json --tiller-namespace kube-system
2018/11/13 13:33:14 INFO: checking if any Helmsman managed releases are no longer tracked by your desired state ...
2018/11/13 13:33:14 INFO: getting helm releases which are managed by Helmsman in namespace [[ kube-system ]].
2018/11/13 13:33:14 VERBOSE: kubectl get secret -n kube-system -l MANAGED-BY=HELMSMAN
2018/11/13 13:33:14 INFO: getting helm releases which are managed by Helmsman in namespace [[ kube-system ]].
2018/11/13 13:33:14 VERBOSE: kubectl get secret -n kube-system -l MANAGED-BY=HELMSMAN
2018/11/13 13:33:14 INFO: no untracked releases found.
2018/11/13 13:33:14 INFO: sorting the commands in the plan based on priorities (order flags) ...
----------------------
2018/11/13 13:33:14 INFO: Plan generated at: Tue Nov 13 2018 20:33:13
DECISION: release [ metrics-server ] is not installed. Will install it in namespace [[ kube-system ]] using Tiller in [ kube-system ] -- priority: 0
2018/11/13 13:33:14 INFO: sorting the commands in the plan based on priorities (order flags) ...
2018/11/13 13:33:14 INFO: Executing the plan ...
2018/11/13 13:33:14 INFO: installing release [ metrics-server ] in namespace [[ kube-system ]] using Tiller in [ kube-system ]
2018/11/13 13:33:14 VERBOSE: helm install stable/metrics-server -n metrics-server --namespace kube-system -f PAPTH/metrics-server.yaml --version 2.0.3 --wait --tiller-namespace kube-system
2018/11/13 13:33:16 Command returned with exit code: . And error message: Error: release metrics-server failed: clusterroles.rbac.authorization.k8s.io is forbidden: User "system:serviceaccount:kube-system:tiller-kube-system" cannot create clusterroles.rbac.authorization.k8s.io at the cluster scope
```

```
kubectl -n kube-system get rolebinding tiller-kube-system-binding -o yaml
apiVersion: rbac.authorization.k8s.io/v1
kind: RoleBinding
metadata:
  creationTimestamp: 2018-11-13T20:32:50Z
  name: tiller-kube-system-binding
  namespace: kube-system
  resourceVersion: "9935"
  selfLink: /apis/rbac.authorization.k8s.io/v1/namespaces/kube-system/rolebindings/tiller-kube-system-binding
  uid: 49e0d38b-e783-11e8-aec0-02112ecfde1a
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: Role
  name: helmsman-tiller
subjects:
- kind: ServiceAccount
  name: tiller-kube-system
  namespace: kube-system
```
```
kubectl -n kube-system get role helmsman-tiller -o yaml
apiVersion: rbac.authorization.k8s.io/v1
kind: Role
metadata:
  annotations:
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"rbac.authorization.k8s.io/v1beta1","kind":"Role","metadata":{"annotations":{},"name":"helmsman-tiller","namespace":"kube-system"},"rules":[{"apiGroups":["","extensions","apps"],"resources":["*"],"verbs":["*"]}]}
  creationTimestamp: 2018-11-13T20:32:49Z
  name: helmsman-tiller
  namespace: kube-system
  resourceVersion: "9931"
  selfLink: /apis/rbac.authorization.k8s.io/v1/namespaces/kube-system/roles/helmsman-tiller
  uid: 49abfa96-e783-11e8-aec0-02112ecfde1a
rules:
- apiGroups:
  - ""
  - extensions
  - apps
  resources:
  - '*'
  verbs:
  - '*'
```

**Comment by regner** [comment]
I am running into the original issue in this. I do how `tillerServiceAccount` set.

```console
2018/12/10 00:13:26 VERBOSE: helm list --tiller-namespace kube-system --tls --tls-ca-cert kube-system-ca.cert --tls-cert kube-system-client.cert --tls-key kube-system-client.key
2018/12/10 00:13:27 ERROR: while waiting for helm Tiller to be ready in namespace [ kube-system ] : Error: configmaps is forbidden: User "system:serviceaccount:kube-system:tiller" cannot list configmaps in the namespace "kube-system"
```

```yaml
namespaces:
  kube-system:
    protected: false
    installTiller: true
    tillerServiceAccount: "tiller"
    caCert: "certs/ca.cert.pem"
    tillerCert: "certs/tiller.cert.pem"
    tillerKey: "certs/tiller.key.pem"
    clientCert: "certs/helm.cert.pem"
    clientKey: "certs/helm.key.pem"
```

**Comment by flyhard** [comment]
I had the same problem. Found out that if I modify the role that is created for tiller I can continue:

Before:

```
---
apiVersion: rbac.authorization.k8s.io/v1
kind: Role
metadata:
  name: helmsman-tiller
  namespace: dev
rules:
  - apiGroups:
      - ""
      - extensions
      - apps
      - rbac.authorization.k8s.io
    resources:
      - '*'
    verbs:
      - '*'
```


after:

```
---
apiVersion: rbac.authorization.k8s.io/v1
kind: Role
metadata:
  name: helmsman-tiller
  namespace: dev
rules:
  - apiGroups:
      - ""
      - extensions
      - apps
    resources:
      - '*'
    verbs:
      - '*'

```

**Comment by sami-alajrami** [comment]
@ironhalo `tillerRole` is now available in v1.7.4
@regner if you are using GKE and getting this error `Error: configmaps is forbidden: User "system:serviceaccount:kube-system:tiller" cannot list configmaps in the namespace "kube-system"`

You will need to give your GCloud user cluster-admin access before running helmsman 
```
kubectl create clusterrolebinding myname-cluster-admin-binding \
  --clusterrole=cluster-admin \
  --user=<your google cloud email>
```
